src/util/system.h

```diff
@@ -365,6 +372,7 @@ class ArgsManager
      * @return command-line argument or default value
      */
     bool GetBoolArg(const std::string& strArg, bool fDefault) const;
+    std::optional<bool> GetBoolArg(const std::string& strArg) const;
```

Comment:
I like the idea of this commit.
My original idea for the future of default arguments was to make ArgsManager handle them internally at registration time, and get rid of the `default` arguments to these functions in that way. But this works.